Userland/Libraries/LibPDF/Shading.cpp

```diff
@@ -644,9 +649,95 @@ PDFErrorOr<void> RadialShading::draw(Gfx::Painter& painter, Gfx::AffineTransform
 using GouraudFunctionsType = Variant<Empty, NonnullRefPtr<Function>, Vector<NonnullRefPtr<Function>>>;
 using GouraudColor = Vector<float, 4>;
 
-PDFErrorOr<void> draw_gouraud_triangle(Gfx::Painter&, NonnullRefPtr<ColorSpace>, GouraudFunctionsType, Array<Gfx::FloatPoint, 3>, Array<GouraudColor, 3>)
+class GouraudPaintStyle final : public Gfx::PaintStyle {
+public:
+    static NonnullRefPtr<GouraudPaintStyle> create(NonnullRefPtr<ColorSpace> color_space, GouraudFunctionsType functions, Array<Gfx::FloatPoint, 3> points, Array<GouraudColor, 3> colors)
+    {
+        return adopt_ref(*new GouraudPaintStyle(move(color_space), move(functions), move(points), move(colors)));
+    }
+
+    // We can't override sample_color() because it doesn't receive a useful origin.
+    // Instead, override `paint()` and pass the origin to similar function.
+    // FIXME: Try changing the signature of sample_color() to receive the actual origin.
```

Comment:
I think overriding `paint` is okay :slightly_smiling_face:, `sample_color` is only meant for simple PaintStyles (most just override `paint`).